### PR TITLE
Feat/location select by algolia

### PR DESCRIPTION
**Pull request by NiksRakevich**
fixes https://github.com/rolling-scopes/rsschool-app/issues/340

### CODE REVIEW

**Comment by mikhama** [comment]
@NiksRakevich pls fix unit-tests

File: client/src/pages/registry/mentor.tsx [hunk 1]
```diff
@@ -243,20 +243,13 @@ function Page(props: Props) {
             <Row gutter={defaultRowGutter}>
               <Col {...defaultColumnSizes}>
                 <Form.Item
+                  help="We need your location for understanding audience and use it for mentor distribution. If you live close to any city from the list, please choose it."
```
Comment (apalchys) [inline client/src/pages/registry/mentor.tsx]: text need to be adapted for mentors registry page. e.g.
`We need your location for students distribution.`

Also `to any city from the list` this line doesn't make any sense anymore

File: client/src/components/LocationSelect.tsx [hunk 2]
```diff
@@ -1,25 +1,43 @@
...
-  [key: string]: any;
+  onChange: Function;
+  location: Location;
 };
 
-type LocationOption = { id?: string; name: string };
-const defaultOption: LocationOption = { id: undefined, name: 'Other' };
-const options: LocationOption[] = [defaultOption].concat(CITIES);
+const getDefaultValue = (location: Location): string =>
+  location && location.countryName !== null ? `${location.countryName}, ${location.cityName}` : '';
```
Comment (apalchys) [inline client/src/components/LocationSelect.tsx]: Suggest simpler syntax:

```typescript
location?.countryName != null ? `${location.countryName}, ${location.cityName}` : '';
```

File: client/src/components/LocationSelect.tsx [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
+      onSuggestions={discardLocation}
+      onClear={discardLocation}
+      onChange={({ suggestion }: any) => {
+        const location = {
+          countryName: suggestion.country,
+          cityName: suggestion.name,
+        } as Location;
+        props.onChange(location);
+      }}
+      placeholder="Write an address here"
```
Comment (apalchys) [inline client/src/components/LocationSelect.tsx]: i think we don't need address, only city. I think it's better to say "Please enter your city" to be consistent with what we display in other controls.

**Comment by apalchys** [comment]
Before you merge, could you please prepare a script for migrating existing data to countryName/cityName ?

**Comment by apalchys** [comment]
additionally, do you plan to remove cities/countries references from the server too?

**Comment by apalchys** [comment]
https://github.com/rolling-scopes/rsschool-app/issues/340

**Comment by NiksRakevich** [comment]
> additionally, do you plan to remove cities/countries references from the server too?

Yes, it need to be removed with fixing features, which use it. (f. e. export score in CSV and other)

**Comment by NiksRakevich** [comment]
> Before you merge, could you please prepare a script for migrating existing data to countryName/cityName ?

Ok, I'll do it

File: client/src/components/LocationSelect.tsx (hunk 2 above)
Comment (NiksRakevich) [reply client/src/components/LocationSelect.tsx]: done

File: client/src/pages/registry/mentor.tsx (hunk 1 above)
Comment (NiksRakevich) [reply client/src/pages/registry/mentor.tsx]: done

**Pull request merged**